---
title: "2022-04-14-tbsP-mutations-AKS"
output: html_document
---

```{r setup, include=FALSE}
# fresh start
rm(list=ls())
knitr::opts_chunk$set(echo = TRUE)

```

#Set up the environment
```{r}
#rm(list = ls()) #comment in if you wish to clear your history prior to starting
library(tidyr)
library(ggplot2)
library(viridis)
library(trackViewer)
library(drawProteins)


#to install tidyr, ggplot2, viridis, simply run the command:
#install.packages("tidyr") #and so on for each package
#to install trackViewer:
#if (!require("BiocManager", quietly = TRUE))
    # install.packages("BiocManager")

# BiocManager::install("trackViewer")



```

## Fig 3A. Cell growth and discovery of TbsP
```{r}
#load in the data
my.data <- read.csv("data/z_Final_growth_data_3.csv")
head(my.data)

#Data rearrangement
my.data1 <- gather(my.data,
                   pyrE:AKS185,
                   key = "Sample",
                   value = "OD600",
                   factor_key = TRUE)
head(my.data1)

#preliminary plot of data
ggplot(my.data1, aes(x = time, y = OD600, color=Sample)) +
  theme(text = element_text(size=25)) +
  geom_line(size=2)
```

```{r}
#final plot of data, Fig 3A

#pdf("Fig3A-supp-growth.pdf", height = 4, width = 6)

ggplot(my.data1, aes(x = time, y = OD600, color = Sample)) +
  #ggtitle('Growthcurves averaged each strain and separated by condition') +
  xlab('Time (h)') + 
 ylab('OD600') +

  geom_smooth() +
  scale_y_continuous(trans = 'log10', limits = c(0.01,1)) +
  ylim (0,1) +
 # facet_grid(environment ~ .) +
  scale_fill_viridis('Sample', discrete = TRUE) +
  scale_color_viridis('Sample', discrete = TRUE) +
  theme(#legend.position = 'none',
    strip.text = element_text(face = "italic",size = 10),
    plot.title = element_text(face='bold', color='#000000', size=12, angle=0),
    axis.title.x = element_text(face='bold', color='#000000', size=10, angle=0),
    axis.title.y = element_text(face='bold', color='#000000', size=10, angle=90),
    legend.title = element_text(face='bold', color='#000000', size=10, angle=0),
    axis.text.y = element_text(face='plain', color='#000000', size=8, angle=0),
    axis.text.x = element_text(face='plain', color='#000000', size=8, angle=0))
    #legend.text = element_text(face='italic', color='#000000', size=10, angle=0))

#dev.off()


```


## Fig 3B. Locations of mutations in tbsP
```{r}


#Generation of genome and mutation location
#SNP <- c(2699503, 2699915, 2700020, 2700030, 2700033, 2700071)

SNP <- c(2699455, 2699503, 2699650, 2699760, 2699805, 2699915, 2699965, 2700020, 2700030, 2700033, 2700071)
sample.gr <- GRanges("chr1", IRanges(SNP, width=1, names=paste0("Mutation at ", SNP)))

#Indicate the mutation types:
#1) deletion, triangle; 2) substitution, circle; 3) repeat expansion, square; 4) addition, diamond
sample.gr$shape <- c("triangle_point_up", "circle", "circle", "square", "diamond", "triangle_point_up", "triangle_point_up", "circle", "circle", "circle", "triangle_point_up")

sample.gr@elementMetadata@listData[["shape"]]

features <- GRanges("chr1", IRanges(c(2699200, 2699406, 2700400), 
                                    width=c(155, 828, 1),
                                    fill=c("grey", "#e69951", "black"),
                                    height=c(0.05, 0.05, 0.02),
                                    names=paste0("block", 1:3)))

pdf("Fig3B-rough-tbsPlocus.pdf")

lolliplot(sample.gr, features)

dev.off()
```




## Drawing protein schematic of TbsP

```{r}
#enter the aa positions of where domain start and end and protein start and end
x <- c(1, 12,54, 123, 141,256,270, 276)
x <- data.frame(x,1) ## 1 is your "height"

#Make the number line, export to Adobe illustrator to add colored lines and labels
pdf("TbsP_visualization.pdf", width = 6, height = 4)

p <- plot(x, type = 'o', pch = '|', ylab = '', xlim = c(1,300), axes = FALSE, xlab = "Amino acid position")
Axis(side = 1)

dev.off()


```









